src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -446,16 +446,42 @@ auto_flash(std::shared_ptr<xrt_core::device> & workingDevice)
   // Collect all indexes of boards need updating
   std::vector<std::pair<unsigned int , DSAInfo>> boardsToUpdate;
 
-  static boost::property_tree::ptree ptEmpty;
-  auto available_shells = pt.get_child(std::to_string(workingDevice->get_device_id()) + ".platform.available_shells", ptEmpty);
+  std::string image_path;
+  if(!image.empty()) {
```

Comment:
From a readability point of view, having a `!empty() `instead of` empty()` makes the code "a little" be more difficult to read.   Is there a reason why this style was used?

And yes, I do know that the condition bodies need to be swapped if you remove the bang '!'